### PR TITLE
add goreleaser and actions config

### PR DESCRIPTION
**Pull request by brandonjbjelland**
## What is this change?

This change uses goreleaser in our actions pipeline when we tag new releases, creating a set of kubectl plugin binaries for different architectures.

## Why make this change?

I want to make the install of the plugin as simple as possible - ideally not requiring users to clone our repo.

While developing the vscode extension functionality to create a Notebook via a single command, users are likely to hit a snag if we're relying on both `kubectl` and our kubectl plugin installed locally. The best experience here is likely handling notebook creation and port forwarding through the vscode extension, but to keep it simple and until we get that user feedback, I'll stick to using the plugin shipped here. The plugin install process needed to be ironed out anyway.

### CODE REVIEW

File: docs/kubectl-plugins.md
Comment (brandonjbjelland) [inline docs/kubectl-plugins.md:28]: TODO(bjb): modify these once I have a clear idea of how the binaries show up in github artifacts

File: .goreleaser.yaml
Comment (samos123) [inline .goreleaser.yaml:13]: Do Mac and Windows work? Did you test those? I've only been testing on Linux so far

**Review by samos123 — COMMENTED** [review]
Big fan of this! Let's also wait for Nick's review but lgtm overall. Left a question regarding testing of Windows and Mac.

File: .goreleaser.yaml
Comment (brandonjbjelland) [reply .goreleaser.yaml:13]: Mac works and I've tested Linux end to end. Let me give it a quick test on Windows to see how it goes.

The `bash -c "curl...` will definitely fail on Windows if it's not WSL. Let me verify Windows on WSL x86_64

Comment (brandonjbjelland) [reply .goreleaser.yaml:13]: works here:

```
Linux raisin 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

Comment (brandonjbjelland) [reply .goreleaser.yaml:13]: Manually downloading the Windows binary worked on W11:

```
PS C:\Users\brand\Downloads\kubectl-open-notebook_Windows_x86_64> .\kubectl-open-notebook.exe
2023/07/10 23:00:14 must provide filename flag (-f) or notebook name
```

File: go.mod
Comment (brandonjbjelland) [inline go.mod:73]: I was dry running the build process and goreleaser had me run `go mod tidy`. LMK if I should undo the bits here.

File: hack/install_kubectl_plugin.sh
Comment (brandonjbjelland) [inline hack/install_kubectl_plugin.sh:8]: A filthy hack because the github api doesn't seem to work unauthenticated even if chatgpt insists it does.

**Review by samos123 — DISMISSED** [review]
lgtm, lets wait for Nick to review since he built the plugins

File: hack/install_kubectl_plugin.sh
Comment (brandonjbjelland) [inline hack/install_kubectl_plugin.sh:15]: best effort here. My mac fails without `sudo` but unsure if that's typical.

File: docs/kubectl-plugins.md
Comment (brandonjbjelland) [inline docs/kubectl-plugins.md:26]: This of course can't be tested fully until we tag a new release. I've been testing it via:

```
wget https://raw.githubusercontent.com/substratusai/substratus/feat/generate-notebook-plugin-release-binaries/hack/install_kubectl_plugin.sh
# change the REPO to point to https://github.com/brandonjbjelland/substratus
# simulate command chain with:
bash -c "$(cat install_kubectl_plugin.sh)"
```

**Review by nstogner — APPROVED** [review]
Quick glance, looks good.

File: hack/install_kubectl_plugin.sh
Comment (nstogner) [inline hack/install_kubectl_plugin.sh:13]: Perhaps we could rely on just curl instead of wget too?

**Comment by brandonjbjelland** [comment]
@samos123 previously approved (dismissed as I reran CI) and @nstogner gave the verbal 🟢 . Merging and testing a release

**Pull request merged**
